### ISSUE TITLE
Make requests span attributes available to samplers for http instrumentations

### ISSUE THREAD

**Issue by lzchen** [open-telemetry/opentelemetry-python-contrib#936]
Like https://github.com/open-telemetry/opentelemetry-python-contrib/pull/931, we should do this for all http instrumentations/

From [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#common-attributes)

**Comment by srikanthccv** [comment]
@codeboten this should supported for the http server instrumentations also right?

**Comment by lzchen** [comment]
@nstawski 
Feel free to comment on the instrumentation issues so we can assign them individually.

**Comment by cheempz** [comment]
@lzchen @srikanthccv could you confirm that this issue can be understood as "Make span attributes available to samplers for http instrumentations", i.e. would apply to both Client and Server attributes that have been called out by the [semantic conventions for HTTP spans](https://github.com/open-telemetry/opentelemetry-specification/blob/ed611c460250198020df249955516f3c5ba7cfbb/specification/trace/semantic_conventions/http.md) to be available at sampling time:

## Common

> Following attributes MUST be provided at span creation time (when provided at all), so they can be considered for sampling decisions:

> http.method

## Client

> Following attributes MUST be provided at span creation time (when provided at all), so they can be considered for sampling decisions:

> http.url
net.peer.name
net.peer.port

## Server

> Following attributes MUST be provided at span creation time (when provided at all), so they can be considered for sampling decisions:

> http.target
http.scheme
net.host.name
net.host.port

This is something we're grappling with and would really help our approach to get a confirmation.  Thanks!

**Comment by srikanthccv** [comment]
>could you confirm that this issue can be understood as "Make HTTP span attributes available to samplers", i.e. would apply to both Client and Server attributes that have been called out by the [semantic conventions for HTTP spans](https://github.com/open-telemetry/opentelemetry-specification/blob/ed611c460250198020df249955516f3c5ba7cfbb/specification/trace/semantic_conventions/http.md) to be available at sampling time:

Yes, the issue description confirms it as there are both HTTP client and server instrumentations listed.

**Comment by cheempz** [comment]
@nstawski is this still on your radar?  We'd love to see these server instrumentation changes!

**Comment by cheempz** [comment]
> It is. I am out on maternity leave until end of April and can pick this up once I am back. Let me know if the timing works.

That would be fantastic @nstawski, thank you :)

**Comment by tammy-baylis-swi** [comment]
Could we add an additional issue for [opentelemetry-instrumentation-asgi](https://github.com/open-telemetry/opentelemetry-python-contrib/tree/main/instrumentation/opentelemetry-instrumentation-asgi)? I'd be happy to take that one.

**Comment by tammy-baylis-swi** [comment]
Hi again, could we add another issue for [opentelemetry-instrumentation-flask](https://github.com/open-telemetry/opentelemetry-python-contrib/tree/main/instrumentation/opentelemetry-instrumentation-flask)? I'd like to take that one too. The change is probably similar to the [recent Django update](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/1730/files).
